NPE in modularity can transformed classes

### Describe the bug

When building my project with `quarkus-jlink`, I get a NPE in `ModularitySteps` line 255, because `bn` is null, with:
```
tc = {TransformedClassesBuildItem$TransformedClass@23383} 
 className = null
 data = null
 fileName = "META-INF/services/io.opentelemetry.sdk.autoconfigure.spi.logs.ConfigurableLogRecordExporterProvider"
```

my app has:
```
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-opentelemetry</artifactId>
        </dependency>
```

the build exception is:
```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.37.2:build (default-build) on project rest-service: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.modular.deployment.ModularitySteps#buildModularityModel threw an exception: java.lang.NullPointerException: Cannot invoke "String.lastIndexOf(int)" because "bn" is null
[ERROR]         at io.quarkus.modular.deployment.ModularitySteps.buildModularityModel(ModularitySteps.java:255)
[ERROR]         at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:811)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1474)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```



### Expected behavior

the build should succeed

### Actual behavior

the build fails

### How to Reproduce?

add `quarkus-opentelemetry` and `quarkus-jlink` to an app, and run `mvn clesn package -DskipTests`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

25

### Quarkus version or git rev

3.37.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvn 3.9

### Additional information

cc @dmlloyd

--- a/extensions/packaging/modular/deployment/src/main/java/io/quarkus/modular/deployment/ModularitySteps.java
+++ b/extensions/packaging/modular/deployment/src/main/java/io/quarkus/modular/deployment/ModularitySteps.java
@@ -251,15 +251,26 @@ public final class ModularitySteps {
         for (Set<TransformedClassesBuildItem.TransformedClass> set : transformedClasses.getTransformedClassesByJar().values()) {
             for (TransformedClassesBuildItem.TransformedClass tc : set) {
                 String bn = tc.getClassName();
+                String fileName = tc.getFileName();
                 String pn;
-                int idx = bn.lastIndexOf('.');
-                if (idx == -1) {
-                    pn = "";
+                if (bn == null) {
+                    // a removed resource rather than a transformed class (see TransformedClassesBuildItem);
+                    // derive the package from the resource path, as is done for generated resources above
+                    int idx = fileName.lastIndexOf('/');
+                    if (idx == -1) {
+                        pn = "";
+                    } else {
+                        pn = fileName.substring(0, idx).replace('/', '.');
+                    }
                 } else {
-                    pn = bn.substring(0, idx);
+                    int idx = bn.lastIndexOf('.');
+                    if (idx == -1) {
+                        pn = "";
+                    } else {
+                        pn = bn.substring(0, idx);
+                    }
                 }
                 // replace any existing entry
-                String fileName = tc.getFileName();
                 byte[] data = tc.getData();
                 if (data == null) {
                     // TODO

